doc/lightning-pay.7.txt

```diff
@@ -30,7 +30,42 @@ RETURN VALUE
 On success, this returns the payment 'preimage' which hashes to the
 'payment_hash' to prove that the payment was successful.
 
-//FIXME:Enumerate errors
+On error, if the error occurred from a node other than the final
+destination, the route table will be updated so that getroute(7)
+should return an alternate route (if any).  An error from the final
+destination implies the payment should not be retried.
+
+The following error codes may occur:
+
+* -1. Catchall nonspecific error.
+* 200. A previous *sendpay* or *pay* is in progress.
+* 201. Already paid with this 'hash' using different amount or
+  destination.
+* 202. Unparseable onion reply. The 'data' field of the error
+  will have a 'onionreply' field, a hex string representation
+  of the raw onion reply.
+* 203. Permanent failure at destination. The 'data' field of
+  the error will be routing failure object.
+* 204. Failure along route; retry a different route. The 'data'
+  field of the error will be routing failure object.
+* 205. Unable to find a route.
+
+A routing failure object has the fields below:
+
+* 'origin_index'. The index of the node along the route that
+  reported the error. 0 for the first hop node, -1 if the local
+  node reported the error.
```

Comment:
Let's add one here, which is far clearer?  eg:

    'origin_index'. The index of the node along the route that reported the error: 0 for the local node,
    1 for the first hop.

Reply:
If you do not mind, also change s/origin_index/erring_index/, then? Internal origin_index counts first hop as 0, I simply extended this to -1 for local node.  Better to reduce confusion for later devs: if they see `origin_index` as return value of `pay`/`sendpay` and then assume that internal `origin_index` created by `unwrap_onionreply` is the same, when in fact the RPC API adds one to the internal value.

Reply:
Agreed, good idea. That is a better name!